### ISSUE TITLE
flake: TestWorkspaceAgent/PostStartup

### ISSUE THREAD

**Issue by spikecurtis** [coder/coder#12939]
Seen at: https://github.com/coder/coder/actions/runs/8630948579/job/23658347582

```
    t.go:99: 2024-04-10 12:15:24.900 [debu]  coderd: GET  host=localhost:37537  path=/api/v2/workspaces/059cc899-201a-4714-9645-194acb81055b  proto=HTTP/1.1  remote_addr=127.0.0.1  start="2024-04-10T12:15:24.369353888Z"  took=531.193913ms  status_code=200  latency_ms=531  request_id=3659f4fd-43c0-4c5b-ad30-486e2f3382be
    coderdtest.go:1040: agent  not connected yet
    agent_test.go:224: 
        	Error Trace:	/home/runner/actions-runner/_work/coder/coder/coderd/coderdtest/coderdtest.go:1018
        	            				/home/runner/actions-runner/_work/coder/coder/cli/agent_test.go:224
        	Error:      	Condition never satisfied
        	Test:       	TestWorkspaceAgent/PostStartup
```

**Comment by spikecurtis** [comment]
This flake happens because `clitest` adds a 15s timeout to CLI commands if there isn't one set explicitly, and for some reason the `test-go-race` suite was super slow.  Running this test locally, without any other tests running takes 300ms on my M1 laptop.  It timed out at 15s in our CI, for a factor of 50+.

**To me, this seems like an unreasonable requirement that we have to build tests that are robust to a 50x span of execution times without flakes.**

Go's blog says the race detector can use 10x the CPU and memory, and it seems approximately true, where enabling it on my M1 laptop takes the test case from 300ms to 2220ms (7.4x).

We could start skipping tests like this one, or manually set longer timeouts, but that's slow and error prone.  I would like us to consider increasing the CPU & Memory of `test-go-race` nodes, or lowering the number of parallel tests (and possibly moving them to nightlies to account for the slower execution time).

@sreya @kylecarbs @ammario

**Comment by ammario** [comment]
> To me, this seems like an unreasonable requirement that we have to build tests that are robust to a 50x span of execution times without flakes.

I agree with this point. We could try using the the 16 core runner while locking the test parallelism at 8. It would be nice to have some statistical method to finding these parameters vs. the endless whac-a-mole  we're playing right now.

**Comment by ammario** [comment]
I have a PoC flaky test wrapper I hope to show you in the coming days.
